src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -769,7 +769,12 @@ public IPAddress MapToIPv4()
         private static byte[] ThrowAddressNullException() => throw new ArgumentNullException("address");
 
         [DoesNotReturn]
-        private static void ThrowSocketOperationNotSupported() => throw new SocketException(SocketError.OperationNotSupported);
+        private void ThrowSocketOperationNotSupported()
+        {
+            string message = IsIPv4 ? "The requested operation is not supported for IPv4 addresses." :
+                                      "The requested operation is not supported for IPv6 addresses.";
```

Comment:
We do not store string message in code. Instead, they should be stored in .resx, with optional format argument and referenced with SR.

You should add a new message in `System.Net.Sockets\src\Resources\Strings.resx` with argument `AddressFamily {0}`.